pandas/core/reshape/concat.py

```diff
@@ -26,8 +28,25 @@
 # Concatenate DataFrame objects
 
 
+@overload
+def concat(
+    objs: Union[Sequence["DataFrame"], Mapping[Hashable, "DataFrame"]],
```

Comment:
IIUC None is a valid key and None is not a subtype of Hashable.

```suggestion
    objs: Union[Sequence["DataFrame"], Mapping[Optional[Hashable], "DataFrame"]],
```

Reply:
None is an instance of Hashable:

```python
In [1]: from typing import Hashable
In [2]: isinstance(None, Hashable)
Out[2]: True
```

So Optional is not needed.

Reply:
however, when typechecking

```python
from typing import Hashable

a: Hashable = None
```
```
$ mypy test.py
test.py:3: error: Incompatible types in assignment (expression has type "None", variable has type "Hashable")
Found 1 error in 1 file (checked 1 source file)
```

Reply:
Hmm, but None *is* hashable (e.g. ``hash(None)`` and ``{None: 1}`` both work). So mypy is specialcasing None, which seems a bit like a bug to me.

Maybe we should have a ```Label = Optional[Hashable]`` in _typing.py? Would seem easier to remember to use that than to define Optional[Hashable] every time?

Reply:
I've added Option. I did not add a Label type to _typing.py Figured it can be done in seperate PR by searching for "Optional[Hashable]"